src/views/Farms/components/FarmTable/Actions/StakedAction.tsx

```diff
@@ -40,8 +41,14 @@ const Staked: React.FunctionComponent<FarmWithStakedValue> = ({ pid, lpSymbol, l
     tokenAddress: token.address,
   })
   const addLiquidityUrl = `${BASE_ADD_LIQUIDITY_URL}/${liquidityUrlPathParts}`
-  const rawStakedBalance = getBalanceNumber(stakedBalance)
-  const displayBalance = rawStakedBalance.toLocaleString()
+
+  const displayBalance = useCallback(() => {
+    const balanceNumberLocale = getBalanceNumber(stakedBalance).toLocaleString()
+    if (stakedBalance.gt(0) && balanceNumberLocale === '0') {
+      return getFullDisplayBalance(stakedBalance).toLocaleString()
+    }
+    return balanceNumberLocale
```

Comment:
This too seems a bit off. I think it would be better to run the checks against the raw value `stakedBalance`, then do the necessary conversion. Is that not possible?

Reply:
We first have to convert it to balanceNumber since stakedBalance doesn't have precision value. But i changed the logic and now we checked against that balance number not the string